addons/website_sale/static/src/js/editor/snippets.options.js

```diff
@@ -35,4 +41,19 @@ options.registry.MegaMenuLayout = options.registry.MegaMenuLayout.extend({
         await this._getTemplate(xmlid);
         await this.selectTemplate(true, xmlid);
     },
+
+    /**
+     * Toggle the state of the button and refresh the mega menu template.
+     *
+     * @returns {void}
+     */
+    async toggleFetchEcomCategories(){
+        /**
+         * The value of the button was set after calling the refresh function,
+         * so we needed to create the fetchEcomCategories variable to be able to update it,
+         * before calling the refresh function.
+         */
```

Comment:
I don't really understand the underlying problem by reading this comment :/ If I'm not mistaken the problem was that the class is added too late right?
```suggestion
         * The class applied by `selectClass` in the web editor it applied after calling this method. Therefore, we need to track the changes manually in a variable to ensure that the right template is rendered by `refreshMegaMenuTemplate`.
```